virtual_ecosystem/models/animals/animal_cohorts.py

```diff
@@ -195,145 +188,418 @@ def die_individual(self, number_dead: int, carcass_pool: CarcassPool) -> None:
+        herbivore's current mass.
+
+        Returns:
+            A float representing the search efficiency rate in [ha/(day*g)].
         """
-        # Calculate the number of individuals that can be eaten based on intake rate
-        # Here we assume predators can consume prey mass equivalent to daily intake
-        number_eaten = min(
-            ceil((predator.intake_rate * predator.individuals) / self.mass_current),
-            self.individuals,
+
+        return sf.alpha_i_k(self.constants.alpha_0_herb, self.mass_current)
+
+    def calculate_potential_consumed_biomass(
+        self, target_plant: Resource, alpha: float
+    ) -> float:
+        """Calculate potential consumed biomass for the target plant.
+
+        This method computes the potential consumed biomass based on the search
+        efficiency (alpha),the fraction of the total plant stock available to the cohort
+        (phi), and the biomass of the target plant.
+
+        Args:
+            target_plant: The plant resource being targeted by the herbivore cohort.
+            alpha: The search efficiency rate of the herbivore cohort.
+
+        Returns:
+            A float representing the potential consumed biomass of the target plant by
+            the cohort [g/day].
+        """
+
+        phi = self.functional_group.constants.phi_herb_t
+        A_cell = 1.0  # temporary
+        return sf.k_i_k(alpha, phi, target_plant.mass_current, A_cell)
+
+    def calculate_total_handling_time_for_herbivory(
+        self, plant_list: Sequence[Resource], alpha: float
+    ) -> float:
+        """Calculate total handling time across all plant resources.
+
+        This aggregates the handling times for consuming each plant resource in the
+        list, incorporating the search efficiency and other scaling factors to compute
+        the total handling time required by the cohort.
+
+        Args:
+            plant_list: A sequence of plant resources available for consumption by the
+            cohort.
+            alpha: The search efficiency rate of the herbivore cohort.
+
+        Returns:
+            A float representing the total handling time in days required by the cohort
+            for all available plant resources.
+        """
+
+        total_handling_t = 0.0
+        phi = self.functional_group.constants.phi_herb_t
+        A_cell = 1.0  # temporary
+        for plant in plant_list:
+            total_handling_t += sf.k_i_k(
+                alpha, phi, plant.mass_current, A_cell
+            ) + sf.H_i_k(
+                self.constants.h_herb_0,
+                self.constants.M_herb_ref,
+                self.mass_current,
+                self.constants.b_herb,
+            )
+        return total_handling_t
+
+    def F_i_k(self, plant_list: Sequence[Resource], target_plant: Resource) -> float:
+        """Refactored method to determine instantaneous herbivory rate on plant k.
```

Comment:
The function name is a obscure. Also the reader of the docstrings does not care it is refactored 😄